src/common.jl

```diff
@@ -0,0 +1,99 @@
+abstract ODEJLAlgorithm <: AbstractODEAlgorithm
+immutable ode23Alg <: ODEJLAlgorithm end
+immutable ode45Alg <: ODEJLAlgorithm end
+immutable ode23sAlg <: ODEJLAlgorithm end
+immutable ode78Alg <: ODEJLAlgorithm end
+
+function solve{uType,tType,isinplace,algType<:ODEJLAlgorithm,F}(prob::AbstractODEProblem{uType,tType,isinplace,F},
+    alg::algType,timeseries=[],ts=[],ks=[];dense=true,save_timeseries=true,
+    saveat=tType[],timeseries_errors=true,reltol = 1e-5, abstol = 1e-8,
+    dtmin = abs(prob.tspan[2]-prob.tspan[1])/1e-9,
+    dtmax = abs(prob.tspan[2]-prob.tspan[1])/2.5,
+    dt = 0.,norm = Base.vecnorm,
+    kwargs...)
+
+    tspan = prob.tspan
+
+    if tspan[end]-tspan[1]<tType(0)
+        error("final time must be greater than starting time. Aborting.")
+    end
+
+    u0 = prob.u0
+
+    Ts = sort(unique([tspan[1];saveat;tspan[2]]))
```

Comment:
Hmm, I'm not a fan of this: it adds unnecessary complexity.  It's not much complexity but feature creep needs to be avoided actively.  That user could easily have done that himself with one or two lines of code.  Now we need to support X lines of code for this instead.

Also, this is wrong when integration is backward in time.  Ah, I see, that is not allowed.  Why not? ODE.jl solver can do it. (I guess I have my feature-creep priorities elsewhere...)